File: RPS.cpp
#include<bits/stdc++.h>
#include<string>

using namespace std;
int main(){
    string yes;
do
{
    string a;
    vector<string> choose={"ROCK", "PAPER","SISSORS"};
    int computer=rand()%3;
    cout<<"choose your pick(ROCK,PAPER,SISSORS): ";
    cin>>a;
    string c=choose[computer];
    /*
    bool c=false; 
    c=find(choose.begin(),choose.end(),a)!=choose.end();
    if(c==false){cout<<"invalid!! please try again;";}*/
    cout<<"============================"<<endl;
    cout<<"Computer choose : "<<c<<endl;
    cout<<"Player choose   : "<<a<<endl;
    cout<<"============================"<<endl;
    
    if(a=="rock")
    {
        if(c=="ROCK"){cout<<"IT IS A TIE"<<endl;}
        if(c=="PAPER"){cout<<"COMPUTER WINS"<<endl;}
        if(c=="SISSORS"){cout<<"PLAYER WINS"<<endl;}
    }
    else if(a=="paper")
    {
        if(c=="PAPER"){cout<<"IT IS A TIE"<<endl;}
        if(c=="SISSORS"){cout<<"COMPUTER WINS"<<endl;}
        if(c=="ROCK"){cout<<"PLAYER WINS"<<endl;}
    }
    else if(a=="sissors")
    {
        if(c=="SISSORS"){cout<<"IT IS A TIE"<<endl;}
        if(c=="ROCK"){cout<<"COMPUTER WINS"<<endl;}
        if(c=="PAPER"){cout<<"PLAYER WINS"<<endl;}
    }

    cout<<"you wanna play again? (Y/N) : ";cin>>yes;
    cout<<endl;
}while(yes=="y");

return 0;}

File: rps.py
import tkinter as tk
from tkinter import ttk
import random

player_score = 0
computer_score = 0

def play_game(player_choice):
    global player_score, computer_score

    choices = ["Rock", "Paper", "Scissors"]
    computer_choice = random.choice(choices)

    result = ""

    if player_choice == computer_choice:
        result = "It's a tie!"
    elif (
        (player_choice == "Rock" and computer_choice == "Scissors") or
        (player_choice == "Paper" and computer_choice == "Rock") or
        (player_choice == "Scissors" and computer_choice == "Paper")
    ):
        result = "You win!"
        player_score += 1
    else:
        result = "Computer wins!"
        computer_score += 1

    result_label.config(text=f"Computer chose {computer_choice}. {result}")
    update_score()

def update_score():
    score_label.config(text=f"Player: {player_score}  |  Computer: {computer_score}")


root = tk.Tk()
root.title("Rock, Paper, Scissors")

# Set the background color
root.configure(bg='#F7DC6F')

title_label = ttk.Label(root, text="Rock, Paper, Scissors", font=("Helvetica", 16), background='#F7DC6F')
title_label.pack(pady=10)

result_label = ttk.Label(root, text="", font=("Helvetica", 12), background='#F7DC6F')
result_label.pack(pady=20)

style = ttk.Style()
style.configure('TButton', font=('Helvetica', 12), background='#3498DB', foreground='#000000')

rock_button = ttk.Button(root, text="Rock", command=lambda: play_game("Rock"))
rock_button.pack(side=tk.LEFT, padx=10)

paper_button = ttk.Button(root, text="Paper", command=lambda: play_game("Paper"))
paper_button.pack(side=tk.LEFT, padx=10)

scissors_button = ttk.Button(root, text="Scissors", command=lambda: play_game("Scissors"))
scissors_button.pack(side=tk.LEFT, padx=10)

# label for the score (player and computers)
score_label = ttk.Label(root, text="Player: 0  |  Computer: 0", font=("Helvetica", 12), background='#F7DC6F')
score_label.pack(pady=10)


root.mainloop()
